File: Algorithms/src/3350. Adjacent Increasing Subarrays Detection II/Code/solution.go
func maxIncreasingSubarrays(nums []int) int {
    n := len(nums)
    if n < 2 {
        return 0
    }

    inc := make([]int, n)
    inc[n-1] = 1
    for i := n - 2; i >= 0; i-- {
        if nums[i] < nums[i+1] {
            inc[i] = inc[i+1] + 1
        } else {
            inc[i] = 1
        }
    }

    feasible := func(k int) bool {
        if k == 0 {
            return true
        }
        for a := 0; a+2*k <= n; a++ {
            if inc[a] >= k && inc[a+k] >= k {
                return true
            }
        }
        return false
    }

    lo, hi, ans := 0, n/2, 0
    for lo <= hi {
        mid := (lo + hi) / 2
        if feasible(mid) {
            ans = mid
            lo = mid + 1
        } else {
            hi = mid - 1
        }
    }
    return ans
}

File: Algorithms/src/3350. Adjacent Increasing Subarrays Detection II/Code/solution.js
var maxIncreasingSubarrays = function (nums) {
  const n = nums.length;
  if (n < 2) return 0;

  const inc = new Array(n).fill(1);
  for (let i = n - 2; i >= 0; --i)
    inc[i] = nums[i] < nums[i + 1] ? inc[i + 1] + 1 : 1;

  const feasible = (k) => {
    if (k === 0) return true;
    for (let a = 0; a + 2 * k <= n; ++a)
      if (inc[a] >= k && inc[a + k] >= k) return true;
    return false;
  };

  let lo = 0,
    hi = Math.floor(n / 2),
    ans = 0;
  while (lo <= hi) {
    const mid = Math.floor((lo + hi) / 2);
    if (feasible(mid)) (ans = mid), (lo = mid + 1);
    else hi = mid - 1;
  }
  return ans;
};
